src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -40,64 +48,104 @@ def _market_stats_channel(self):
     def _grid_fees_channel(self):
         return f"{self.channel_prefix}/grid_fees"
 
-    def sub_to_area_event(self):
-        self.redis_db = self.area.config.external_redis_communicator
-        self.redis_db.sub_to_multiple_channels({
+    @staticmethod
+    def _get_transaction_id(payload):
+        data = json.loads(payload["data"])
+        if "transaction_id" in data and data["transaction_id"] is not None:
+            return data["transaction_id"]
+        else:
+            raise ValueError("transaction_id not in payload or None")
+
+    def _register(self, payload):
+        self._connected = register_area(self.redis_com, self.channel_prefix, self.connected,
+                                        self._get_transaction_id(payload),
+                                        area_uuid=self.area.uuid)
+
+    def _unregister(self, payload):
+        self._connected = unregister_area(self.redis_com, self.channel_prefix, self.connected,
+                                          self._get_transaction_id(payload))
+
+    def sub_to_external_channels(self):
+        self.redis_com = self.area.config.external_redis_communicator
+        sub_channel_dict = {
             f"{self.channel_prefix}/market_stats": self.market_stats_callback,
             f"{self.channel_prefix}/dso_market_stats": self.dso_market_stats_callback,
-            f"{self.channel_prefix}/grid_fees": self.set_grid_fees_callback
-        })
+            f"{self.channel_prefix}/grid_fees": self.set_grid_fees_callback,
+            f"{self.channel_prefix}/register_participant": self._register,
+            f"{self.channel_prefix}/unregister_participant": self._unregister}
+        if self.area.config.external_redis_communicator.is_enabled:
+            self.aggregator = self.area.config.external_redis_communicator.aggregator
+        self.redis_com.sub_to_multiple_channels(sub_channel_dict)
 
     def market_stats_callback(self, payload):
         market_stats_response_channel = f"{self.channel_prefix}/response/market_stats"
-        payload_data = json.loads(payload["data"])
+        payload_data = json.loads(payload["data"]) \
+            if isinstance(payload["data"], str) else payload["data"]
         ret_val = {"status": "ready",
+                   "area_uuid": self.area.uuid,
                    "command": "market_stats",
                    "market_stats":
                        self.area.stats.get_market_stats(payload_data["market_slots"]),
                    "transaction_id": payload_data.get("transaction_id", None)}
-        self.redis_db.publish_json(market_stats_response_channel, ret_val)
+        if self.is_aggregator_controlled:
+            ret_val.pop("transaction_id")
```

Comment:
This would not be needed IMO if the transaction_id was popped and not `get` from the payload_data 3 lines above.

Reply:
So you are suggest to rather add it to the retval for the case when the market is not aggregator_controlled and not pop it here?

Reply:
Ah I confused the logic here, so this is the inverse than what I thought, transaction_id is always populated and deleted if we have an aggregator and transaction_ids do not make sense for individual commands. 
I think it would be cleaner though, if the transaction_id was not populated at all in the aggregator case and was only populated in individual commands. This is a matter of personal preference though and YMMV, therefore feel free to ignore this comment and its follow-up. It is fine as it is even now

Reply:
Oh no, you convinced me actually. It would be more clearer your way. Will change this, thanks!